### PR TITLE
query: query remaining peers when NumRetries is 0

### PR DESCRIPTION
**Pull request by wpaulino**
NumRetries signals that we should retry all of our peers a certain amount of times. Before this commit, if set to 0, we'd skip over all of our remaining peers because peerTries would return a 0 value if the peer had yet to be tried. We fix this by ensuring that we try all peers _once_ regardless of what NumRetries is set to.

### CODE REVIEW

File: query.go [hunk 1]
```diff
@@ -662,21 +662,31 @@ checkResponses:
...
+				// If we've yet to try this peer, we'll make
+				// sure to do so. If we've exceeded the number
+				// of tries we should retry this peer, then
+				// we'll skip them.
+				numTries, ok := peerTries[peer.Addr()]
+				if ok && numTries >= qo.numRetries {
+					continue
 				}
+
+				queryPeer = peer
```
Comment (halseth) [inline query.go]: Is it safe to drop the `curPeer := curPeer`?

Comment (wpaulino) [reply query.go]: Should be, there are no goroutines within the loop.

File: query.go [hunk 2]
```diff
@@ -662,21 +662,31 @@ checkResponses:
...
-					peerTries[queryPeer.Addr()]++
-					queryPeer.subscribeRecvMsg(subscription)
-					queryPeer.QueueMessageWithEncoding(
-						queryMsg, nil, qo.encoding,
-					)
-					break
+			for _, peer := range s.Peers() {
+				// If the peer is no longer connected, we'll
+				// skip them.
+				if !peer.Connected() {
```
Comment (cfromknecht) [inline query.go]: Missing `peer != nil` check

Comment (wpaulino) [reply query.go]: Why is the check needed?

Comment (cfromknecht) [reply query.go]: To prevent panics, check the history

Comment (wpaulino) [reply query.go]: Are you referring to https://github.com/lightninglabs/neutrino/commit/9a0709b914d3ea43113806d8b8aaf6f87e06bf44? That seems different -- we query for a peer given their address. The `nil` check makes sense there to me as the peer can no longer exist, but here `Peers` is just returning the currently active peers.

Comment (cfromknecht) [reply query.go]: Hmm, actually i think I’m recalling a different case where checking nilness is necessary. Why is used atm?

Comment (wpaulino) [reply query.go]: I think it was left over from some shadowing changes, see https://github.com/lightninglabs/neutrino/commit/fd24569c15b5960f51dc3f336954b8406a2beac1.

Comment (cfromknecht) [reply query.go]: Ahh okay, thanks for clarifying. Clear to me now that it is not needed

**Pull request merged**